[Zephyr] Fix race condition in NoSelect event loop implementation

ArekBalysNordic opened the pull request:
#### Summary

When using the ZephyrNoSelect platform manager implementation, timers scheduled from threads other than the event loop thread would fire too quickly, before the scheduling operation completed.
For example, when adding a dynamic Matter bridge endpoint, the subscription report would be sent before the endpoint was fully created, resulting in incomplete data being transmitted.
    
The original implementation immediately woke up the event loop when _StartChipTimer was called, causing timers to fire before the calling thread finished its operation.
    
Implemented a delayed wake-up mechanism using a Zephyr kernel timer:
    
- Replace direct k_msgq_get with k_poll on both message queue and
    semaphore, allowing the event loop to be woken by multiple sources.
- In _StartChipTimer, instead of immediately waking the event loop,
    schedule a 1ms delayed wake-up using a Zephyr kernel timer. This
    gives the calling thread time to complete its operation before
    the event loop processes the timer.
- The timer callback runs in ISR context and uses a work queue to
    safely signal the event loop semaphore.
- Remove polling - the event loop now sleeps until:
      1. An event is posted to the message queue
      2. The semaphore is signaled (from timer or ISR)
      3. The timeout for the next pending timer expires

#### Testing

Tested locally on nRF5340 DK and nRF54LM20 DK with the Bridge device type and the `chip_system_config_use_sockets` enabled. Other reporting tests are covered by the unit tests.

#### Readability checklist

The checklist below will help the reviewer finish PR review in time and keep the
code readable:

-   [ ] PR title is
        [descriptive](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html#title-formatting)
-   [ ] Apply the
        [_“When in Rome…”_](https://project-chip.github.io/connectedhomeip-doc/style/CODING_STYLE_GUIDE.html)
        rule (coding style)
-   [ ] PR size is short
-   [ ] Try to avoid "squashing" and "force-update" in commit history
-   [ ] CI time didn't increase

See: [Pull Request Guidelines](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=42373) <br/>All committers have signed the CLA.

kkasperczyk-no reviewed (approved):
LGTM

andy31415 commented on src/app/util/attribute-storage.cpp:
```diff
@@ -390,7 +390,9 @@ CHIP_ERROR emberAfSetDynamicEndpointWithEpUniqueId(uint16_t index, EndpointId id
 
     // Now enable the endpoint.
-    emberAfEndpointEnableDisable(id, true);
+    VerifyOrReturnError(chip::DeviceLayer::SystemLayer().ScheduleLambda([id]() { emberAfEndpointEnableDisable(id, true); }) ==
```
same here: update the comment to say why schedulelambda is important instead of direct inline.

ArekBalysNordic committed:
[Zephyr] Fix race condition in NoSelect event loop implementation

When using the ZephyrNoSelect platform manager implementation,
timers scheduled from threads other than the event loop thread
would fire too quickly, before the scheduling operation completed.
For example, when adding a dynamic Matter bridge endpoint,
the subscription report would be sent before the endpoint was
fully created, resulting in incomplete data being transmitted.

The original implementation immediately woke up the event loop when
_StartChipTimer was called, causing timers to fire before the calling
thread finished its operation.

Implemented a delayed wake-up mechanism using a Zephyr kernel timer:

- Replace direct k_msgq_get with k_poll on both message queue and
semaphore, allowing the event loop to be woken by multiple sources.

- In _StartChipTimer, instead of immediately waking the event loop,
schedule a 1ms delayed wake-up using a Zephyr kernel timer. This
gives the calling thread time to complete its operation before
the event loop processes the timer.

- The timer callback runs in ISR context and uses a work queue to
safely signal the event loop semaphore.

- Remove polling - the event loop now sleeps until:
  1. An event is posted to the message queue
  2. The semaphore is signaled (from timer or ISR)
  3. The timeout for the next pending timer expires

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>
